.github/workflows/main.yml

```diff
@@ -0,0 +1,52 @@
+name: CI
+on: 
+  pull_request:
+  push:
+    branches:
+      - main
+      - add_GHA_based_CI
```

Comment:
FYI, another way you can do this is to add the `workflow_dispatch:` key. Then you'll be able to trigger the workflow on any branch manually from the GitHub UI.

Reply:
Also, if you are contributing to a project, you can always make a PR to your own project first as an easy way to test the CI. So if I make a PR to `pypa/build`'s `main` branch, I could first make a PR to my own `henryiii/build` fork's `main` branch to trigger the CI without anyone seeing the result (or using `pypa`'s CI bandwidth).

Just a handy trick sometimes.